data_analysis/plan_choice_helpers/methane_plan_helpers.py

```diff
@@ -0,0 +1,254 @@
+            | full_df.apply(is_simple_night_uncontrolled, axis=1)
+        )
+    )
+    full_df = full_df[filters]
+
+    # Log number of unique retailer and network locations
+    retailer_locs = full_df["Retailer location name"].unique()
+    network_locs = full_df["Network location names"].unique()
+    logger.info("There are %s unique retailer locations", len(retailer_locs))
+    logger.info("There are %s unique network locations", len(network_locs))
+
+    full_df.reset_index(drop=True, inplace=True)
+    full_df.drop_duplicates(inplace=True)
+
+    return full_df
+
+
+def plot_subset(df, edb=None, hue_column=None, output_dir="scatterplots"):
+    """
+    Plot an EDB-specific subset of the DataFrame.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The DataFrame containing the data to plot.
+
+    edb : str or None, optional
+        The EDB name to filter the data by.
+        If None, all EDBs are included. Default is None.
+
+    hue_column : str or None, optional
+        The column to use for coloring the data points.
+        If None, no coloring is applied. Default is None.
+
+    output_dir : str, optional
+        The output directory to save the plot image to.
+        Default is "scatterplots".
+
+    Returns
+    -------
+    None
+    """
+    subset = df.copy()
+    if edb is not None:
+        subset = subset[subset["EDB"] == edb]
+    subset["1/10 x Daily charge"] = subset["Daily charge"] / 10
+    scatterplot_vars = [
+        "All inclusive",
+        "Day",
+        "Uncontrolled",
+        "Controlled",
+        "Night",
+        "1/10 x Daily charge",
+    ]
+    plot_height = 1
+    plot_aspect = 2
+    if (
+        hue_column is not None
+        and hue_column in subset.columns
+        and subset[hue_column].notnull().any()
+    ):
+        g = sns.pairplot(
+            subset,
+            vars=scatterplot_vars,
+            hue=hue_column,
+            palette="Set1",
+            plot_kws={"alpha": 0.6},
+            height=plot_height,
+            aspect=plot_aspect,
+        )
+    else:
+        g = sns.pairplot(
+            subset,
+            vars=scatterplot_vars,
+            plot_kws={"alpha": 0.6},
+            height=plot_height,
+            aspect=plot_aspect,
+        )
+    for ax in g.axes.flatten():
+        if ax is not None:
+            ax.set_xlim(0, 0.5)
+            ax.set_ylim(0, 0.5)
+    g.fig.suptitle(edb if edb else "All EDBs", fontsize=16, y=0.98)
+    os.makedirs(output_dir, exist_ok=True)
+    filename = f"{output_dir}/{edb.replace(' ', '_') if edb else 'all'}.png"
+    plt.savefig(filename)
+    plt.close()
+
+
+def load_gas_using_household_energy_usage_profile():
+    """
+    Load household fuel usage profile reflecting the 'default'
+    answers. These correspond to a typical gas using household.
+
+    Returns:
+    YearlyFuelUsageProfile object representing the household's
+    fuel usage profile.
+    """
+    household_profile = HouseholdAnswers(
```

Comment:
There is a similar but distinct function which is for an electrified household and is used for optimizing electricity plans. 

In an earlier PR you made comments about 'default' profile (which corresponded to the 'electrified' household profile, and which pre-dated the decision to optimize natural gas plans in the same way as electricity) that made me realize it would be better to move the definitions of these profiles closer to where they are used.

Based on that realization I want this function to be completely explicit about all of the settings. The docstrings need to be fixed to explain these functions better - have updated them now.